lib/src/js/polymer_base_dart.html

```diff
@@ -75,12 +75,26 @@
       // Secret hook into property changes. Pretty hacky but its more efficient
       // than using a JsProxy object for the element.
       _propertyChanged: function(path, newValue, oldValue) {
+        // resolving with keys for array - is it a bug or a feature ?
         var parts = path.split('.');
-        var thisArg = parts.length == 1 ?
-            this : this.get(parts.slice(0, parts.length - 1));
+        var thisArg =  this;
+        for (var i=0;i<parts.length-1;i++) {
```

Comment:
Yes. I couldn't believe it myself. I had to double check it to be sure I wasn't missing  something and I'm still not convinced 100%.
But the test case clearly fails because of that. You try to update 'myThings.1.field' then it changes the path in 'my things.2.field' but element 2 doesn't exist and fails ...